util-images/watch-list/main.go

```diff
@@ -0,0 +1,103 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	v1 "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+var (
+	kubeconfig                  string
+	targetNamespace             string
+	informerCount               int
+	testTimeout                 time.Duration
+	enableWatchListAlphaFeature bool
+)
+
+func main() {
+	registerFlags()
+	flag.Parse()
+
+	klog.Info("The test binary started with the following arguments:")
+	flag.VisitAll(func(f *flag.Flag) {
+		klog.Infof("  -%s=%v (%s)\n", f.Name, f.Value, f.Usage)
+	})
+
+	if enableWatchListAlphaFeature {
+		os.Setenv("ENABLE_CLIENT_GO_WATCH_LIST_ALPHA", "true")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	config.AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json"
+	config.ContentType = "application/vnd.kubernetes.protobuf"
+	klog.Infof("The following Kubernetes client config will be used\n%v", config.String())
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	wait.UntilWithContext(ctx, func(ctx context.Context) {
+		ts := time.Now()
+		ctxInformer, cancelInformers := context.WithCancel(ctx)
+		defer cancelInformers()
+
+		klog.Infof("Starting %d secret informers for targetNamespace = %s", informerCount, targetNamespace)
+		informersSynced := startInformersFor(ctxInformer, client, informerCount, targetNamespace)
+
+		klog.Info("Waiting for the secrets informers to synced")
+		cache.WaitForCacheSync(ctx.Done(), informersSynced...)
+		klog.Infof("All %v secret informers synced, time needed = %v", len(informersSynced), time.Now().Sub(ts))
+	}, 5*time.Second)
+
+	klog.Info("Exiting the test app")
+}
+
+func registerFlags() {
+	klog.InitFlags(flag.CommandLine)
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "path to kubeconfig.")
+	flag.StringVar(&targetNamespace, "namespace", "huge-secrets-1", "namespace that host secrets to list. If empty a default (huge-secrets-1) value will be used.")
+	flag.IntVar(&informerCount, "count", 4, "the number of informers per targetNamespace to run. If empty a default (4) value will be used.")
+	flag.DurationVar(&testTimeout, "timeout", time.Minute, "timeout duration for the test")
```

Comment:
I think that setting binary-level timeout is not compatible with the current approach in https://github.com/kubernetes/perf-tests/pull/2255 where you delete pods after sleep period.

What will be happening is that the binary will be restarting every 1m during the $testDuration (5m) period.

Reply:
oh, this is just a default timeout, I was planing to set a new one directly from the test.

Reply:
here: https://github.com/kubernetes/perf-tests/pull/2255/files#diff-58ef704e64da515ecf846e260d5440f01f331c7b67741b51368887d99d7b8d16R22